src/components/GLStyleService.js

```diff
@@ -0,0 +1,122 @@
+goog.provide('ga_gl_style_service');
+(function() {
+  var module = angular.module('ga_gl_style_service', []);
+
+  module.provider('gaGLStyle', function() {
+    this.$get = function($http, $q, $window) {
+      var GLStyle = function() {
+        this.styleJSONCache = null;
+
+        this.spriteJSONCache = null;
+
+        this.filters = [];
+
+        this.edits = [];
+
+        this.get = function(styleUrl) {
+          var that = this;
+          var defer = $q.defer();
+          $http.
+              get(styleUrl, {
+                cache: true
+              }).
+              then(
+                  function(response) {
+                    that.styleJSONCache = response.data;
```

Comment:
So we keep only the one glStyle in cache ?  when we change the bg layer it will be replaced, am I right?